cameo/api/adapter.py

```diff
@@ -0,0 +1,181 @@
+    try:
+        clean_id = clean_bigg_id(metanetx.mnx2bigg[mnx_id])
+        return model.metabolites.get_by_id(clean_id + compartment)
+    except KeyError:
+        try:
+            return model.metabolites.get_by_id(mnx_id + compartment)
+        except KeyError:
+            pass
+
+
+def contains_carbon(metabolite):  # TODO: use method from Metabolite class when this change is merged
+    if not metabolite.formula:
+        raise ValueError("No formula for metabolite {}, it's unknown if there is carbon in it")
+    return 'C' in metabolite.elements
+
+
+def find_metabolite_info(met_id):
+    """Find chemical formula of metabolite in metanetx.chem_prop dictionary
+
+    Parameters
+    ----------
+    met_id : string
+        string of format "<metabolite_id>_<compartment_id>", where <metabolite_id> is a BIGG id or a Metanetx id
+
+    Returns
+    -------
+    pandas row or None
+
+    """
+    met_id = met_id[:-2]
+    try:
+        if met_id in metanetx.chem_prop.index:
+            return metanetx.chem_prop.loc[met_id]
+        return metanetx.chem_prop.loc[metanetx.all2mnx['bigg:' + met_id]]
+    except KeyError:
+        return None
+
+
+class ModelModification(object):
+    """
+    Base model modification class, providing methods for adding adapter and exchange reactions for new metabolites
+    """
+    model = None
+    added_reactions = None
+
+    def create_exchange(self, gene_name, metabolite):
+        """For given metabolite A_c from c compartment, create:
+        a) corresponding metabolite A_e from e compartment;
+        b) adapter reaction A_c <--> A_e
+        c) exchange reaction A_e -->
+
+        Parameters
+        ----------
+        gene_name : string
+            metabolite id in format <bigg_id>_c, f.e. Nacsertn_c
+        metabolite : Metabolite
+            gene associated with the metabolite
+
+        Returns
+        -------
+
+        """
+        exchange_metabolite = Metabolite(metabolite.id.replace('_c', '_e'), formula=metabolite.formula, compartment='e')
+        self.add_adapter_reaction(metabolite, exchange_metabolite, gene_name)
+        self.add_exchange_reaction(exchange_metabolite)
+
+    def add_exchange_reaction(self, metabolite):
+        """Add exchange reaction "A --> " for given metabolite A
+         If reaction exists, log and pass
+
+        Parameters
+        ----------
+        metabolite : basestring
+            metabolite id in format <bigg_id>_<compartment_id>, f.e. Nacsertn_c
+
+        Returns
+        -------
+
+        """
+        try:
+            logger.debug('Add exchange reaction for metabolite: {}'.format(metabolite.id))
+            exchange_reaction = self.model.add_exchange(metabolite, prefix='EX_')
+            self.added_reactions.add(exchange_reaction.id)
+        except ValueError:
+            logger.debug('Exchange reaction exists for metabolite {}'.format(metabolite.id))
+
+    def add_adapter_reaction(self, metabolite, existing_metabolite, gene_name):
```

Comment:
That I understood, but adapter reactions are just virtual reactions that connect a metabolite from a model to another model/database so that they are the same right? That reaction is not a representation of an existing reaction carried by a protein so it should not be annotated with a gene reaction rule.